src/aiida/storage/sqlite_dos/backend.py

```diff
@@ -225,3 +234,36 @@ def _get_mapper_from_entity(entity_type: 'EntityTypes', with_pk: bool):
         mapper = inspect(model).mapper  # type: ignore[union-attr]
         keys = {key for key, col in mapper.c.items() if with_pk or col not in mapper.primary_key}
         return mapper, keys
+
+    def _backup(
+        self,
+        dest: str,
+        keep: Optional[int] = None,
+    ):
+        try:
+            backup_manager = backup_utils.BackupManager(dest, keep=keep)
+            backup_manager.backup_auto_folders(lambda path, prev: self._backup_storage(backup_manager, path, prev))
+        except backup_utils.BackupError as exc:
+            raise exceptions.StorageBackupError(*exc.args) from exc
+
+    def _backup_storage(
```

Comment:
nit: I would move this function before `_backup()` since it has a nice docstring that explains what is going on (helpful when reading from top to bottom).

Reply:
Hmm, I did it this way because `_backup` is the actual abstractmethod that is implemented. The `_backup_storage` is purely an internal method that needed to be separate because of the way `_backup` is implemented. I think I'd rather just add a proper method to `_backup` as well.